runtime/src/qio/auxFilesys/hdfs/qio_plugin_hdfs.c

```diff
@@ -138,6 +138,25 @@ qioerr hdfs_writev(void* fl, const struct iovec* iov, int iovcnt, ssize_t* num_w
   return err_out;
 }
 
+/* 
+   the HDFS3 variable is provided by build system found in files:
+
+   $CHPL_HOME/runtime/etc/Makefile.auxFilesys
+   $CHPL_HOME/runtime/src/qio/auxFilesys/hdfs/Makefile.share
+
+   this slice of code is a simple implementation of an hdfs function 
+   is not currently supported by libhdfs3
+*/
+
+#ifdef HDFS3
+
+tSize hdfsPread(hdfsFS fs, hdfsFile file, tOffset position, void* buffer, tSize length) {
+   hdfsSeek(fs, file, position);
+   return hdfsRead(fs, file, buffer, length);
```

Comment:
The 3rd option is the most straight forward - it's been coded up. I'll do a build, test, and aim to make a push tomorrow!